HOOKS.md

```diff
@@ -0,0 +1,59 @@
+# Hooks
+
+The ansible installer allows for operators to execute custom tasks during
+specific operations through a system called hooks. Hooks allow operators to
+provide files defining tasks to execute before and/or after specific areas
+during installations and upgrades. This can be very helpful to validate
+or modify custom infrastructure when installing/upgrading OpenShift.
+
+It is important to remember that when a hook fails the operation fails. This
+means a good hook can run multiple times and provide the same results. A great
+hook is idempotent.
+
+**Note**: There is currently not a standard interface for hooks. In the future
+a standard interface will be defined and any hooks that existed previously will
+be updated to meet the new standard.
+
+## Using Hooks
+
+Hooks are defined in the ``hosts`` inventory file under the ``OSEv3:vars``
+section.
+
+Each hook should point to a yaml file which defines Ansible tasks. Best
+practice suggests using absolute paths to the hook file to avoid any ambiguity.
```

Comment:
Probably needs to be clarified to state that the hooks are an included tasks file rather than a playbook.

Also, needs to say that they are run against each of the master hosts, so tasks that would need to be run against a different host will need to use `delegate_to` or `local_action`